drivers/sub8_actuator_driver/scripts/actuator_driver.py

```diff
@@ -59,18 +59,19 @@ def __init__(self, port, baud=9600):
         rospy.Service('~actuate', SetValve, self.got_service_request)
         rospy.Service('~actuate_raw', SetValve, self.set_raw_valve)
 
-        rospy.spin()
-        # r = rospy.Rate(1)  # hz
-        # while not rospy.is_shutdown():
-        #     # Heartbeat to make sure the board is still connected.
-        #     r.sleep()
-        #     self.ping()
-        #     if not self.parse_response(None):
-        #         rospy.logwarn("The board appears to be disconnected, trying again in 5 seconds.")
-        #         self.thruster_out_alarm.raise_alarm(
-        #             problem_description='The board appears to be disconnected.'
-        #         )
-        #         rospy.sleep(5)
+        #rospy.spin()
+        # Might comment this whole deal out once it all works for sure.
+        r = rospy.Rate(.2)  # hz
+        while not rospy.is_shutdown():
+            # Heartbeat to make sure the board is still connected.
+            r.sleep()
+            self.ping()
```

Comment:
Yep, ran into this problem with Congregator comms where spamming data could cause packet loss. I was going to just not ping and use service calls as pings but locking the thread will be much much better. Will implement tomorrow and recommit with that to this PR.